src/parser.ts

```diff
@@ -37,6 +37,20 @@ export function parse(
   usedNames = new Set<string>()
 ): AST {
   if (isPrimitive(schema)) {
+    if (options.booleanSchemas) {
+      if (schema === true) {
```

Comment:
Please move this new logic into the `parseLiteral` function.

Reply:
In addition, this is sort of buggy. Enabling boolean requires the distinction between parsing boolean literals and boolean schemas. Enum schemas with boolean values requires parsing the boolean value as a literal, as an example.